Add OIE Plugin Guide documentation (#5)
OpenIntegrationEngine/docs-website: docs/engine/plugins.md

For environments where UI-based installation is not feasible, you can install plugins manually:

1. Stop the OIE Server
2. Extract the plugin `.zip` file
3. Copy the extracted plugin folder to the `extensions/` directory in your OIE installation:
   ::: code-group
   ```bash [Linux/macOS]
   cp -r my-plugin /opt/oie/extensions/
   ```
   ```powershell [Windows]
   Copy-Item -Recurse my-plugin "C:\Program Files\OIE\extensions\"
   ```
   :::
4. **(Linux/macOS only)** Ensure proper file permissions so the OIE process can read the plugin files. The plugin files should be owned by the same user that runs OIE:
   ::: code-group
   ```bash [Linux/macOS]
   # Replace 'oie' with the user account that runs the OIE Server
   chown -R oie:oie /opt/oie/extensions/my-plugin
   ```
   :::
   ::: tip
   OIE does not create a dedicated system user during installation. It runs as the user who starts the service. Check which user owns your OIE installation directory or runs the OIE process.
   :::
5. Start the OIE Server

### Installing in the OIE Docker Container

When running OIE in a Docker container, you have several options for installing plugins:

#### Option 1: Volume Mount (Recommended)

> Mount a host directory containing plugins to the container's extensions directory:

Reviewer: ```suggestion
Mount a host directory containing plugin zip files to the container's `custom-extensions` directory. The entrypoint script will unzip and install them to the `extensions` directory prior to launching the server:
```

```bash
docker run -d \

>   -v /path/to/local/extensions:/opt/oie/extensions \

Reviewer: ```suggestion
  -v /path/to/local/custom-extensions:/opt/engine/custom-extensions \
```

  openintegrationengine/engine:latest
```

#### Option 2: Custom Dockerfile

Create a custom Docker image with plugins pre-installed:

```dockerfile
FROM openintegrationengine/engine:latest

COPY my-plugin.zip /tmp/
RUN unzip /tmp/my-plugin.zip -d /opt/oie/extensions/ && \
    rm /tmp/my-plugin.zip
```

Build and run:

```bash
docker build -t oie-with-plugins .
docker run -d oie-with-plugins
```

#### Option 3: Docker Compose

Using Docker Compose with volume mounts:

```yaml
version: '3.8'
services:
  oie:
    image: openintegrationengine/engine:latest
    volumes:

>       - ./extensions:/opt/oie/extensions

Reviewer: ```suggestion
      - ./custom-extensions:/opt/engine/custom-extensions
```
